x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/NamedDateTimeFunction.java

```diff
@@ -33,14 +31,16 @@
     }
 
     @Override
-    public ScriptTemplate scriptWithField(FieldAttribute field) {
-        return new ScriptTemplate(
-                formatTemplate(format(Locale.ROOT, "{sql}.%s(doc[{}].value, {})",
-                        StringUtils.underscoreToLowerCamelCase(nameExtractor.name()))),
-                paramsBuilder()
-                  .variable(field.name())
-                  .variable(zoneId().getId()).build(),
-                dataType());
+    public ScriptTemplate asScript() {
+        ParamsBuilder params = paramsBuilder();
```

Comment:
Minor nitpick but you could declare `ParamsBuilder` after the template declaration since it is not used before then.